File: Graphics/line2.pl
#: line2.pl
#: Copyright (c) 2006 Agent Zhang
#: 2006-03-17 2006-03-17

use strict;
use warnings;
use GD::Simple;

my $im = GD::Simple->new(40,50);
my $red = $im->colorAllocate(255,0,0); 

#$img->moveTo(0, 0);
#$img->lineTo(20, 30);
draw_line($im, 0, 0, 20, 30);

binmode \*STDOUT;
print $im->png;

sub draw_line {
    my ($im, $x_O, $y_O, $x_A, $y_A) = @_;
    my ($x, $y) = ($x_O, $y_O);
    $im->setPixel($x, $y, $red);
    while ($x != $x_A || $y != $y_A) {
        my $F_M = $y * $x_A - $y_A * $x;
        if ($F_M < 0) {
            $im->setPixel($x, ++$y, $red);
        } elsif ($F_M > 0) {
            $im->setPixel(++$x, $y, $red);
        } else {
            $im->setPixel(++$x, ++$y, $red);
        }
    }
    #$im->setPixel($x, $y, $red);
}

File: Graphics/GD++/t/line2.cpp
//: line2.cpp
//: Copyright (c) 2006 Agent Zhang
//: 2006-04-10 2006-04-10

#include <gdpp.h>

void draw_line(GD::Simple im, int x_0, int y_0, int x_A, int y_A, GD::Color color) {
    int x = x_0;
    int y = y_0;
    im.setPixel(x, y, color);
    while (x != x_A || y != y_A) {
        int F_M = y * x_A - y_A * x;
        if (F_M < 0) {
            im.setPixel(x, ++y, color);
        } else if (F_M > 0) {
            im.setPixel(++x, y, color);
        } else {
            im.setPixel(++x, ++y, color);
        }
    }
}

int main() {
    GD::Simple im(40, 50);
    GD::Color red = im.colorAllocate(255, 0, 0); 

    draw_line(im, 0, 0, 20, 30, red);

    im.as_png("line2.png");
    return 0;
}
